sonar-plugin-api/src/test/java/org/sonar/api/compute/MetricsDefinitionTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.api.compute;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MetricsDefinitionTest {
+
+  MetricsDefinition.NewMetricContext newMetricContext = new MetricsDefinition.NewMetricContext();
+
+  @Test
+  public void define_metric() throws Exception {
+    assertThat(newMetricContext.getMetrics()).isEmpty();
+
+    newMetricContext.createNewMetric()
+      .setKey("key")
+      .setName("name")
+      .createMeasureComputer()
+      .setInputMetricKeys("key1", "key2")
+      .setMeasureComputer(new MetricsDefinition.MeasureComputer() {
+        @Override
+        public void compute(MetricsDefinition.MeasureComputerContext context) {
+
+        }
+      })
+      .done()
+      .done();
```

Comment:
to solve this, you could simply replace `createMeasureComputer` with `setMeasureComputer` and provide `MeasureComputer.newMeasureComputerBuilder`

(you could rename `createNewMetric` to `newMetricBuilder` and `done` could be called `build` which more common)